pages/curriculum/[lesson].tsx

```diff
@@ -1,27 +1,41 @@
 import * as React from 'react'
 import { useQuery } from '@apollo/react-hooks'
 import { useRouter } from 'next/router'
+import useSession from '../../helpers/useSession'
 import Layout from '../../components/Layout'
 import LessonTitleCard from '../../components/LessonTitleCard'
 import Alert from '../../components/Alert'
 import ChallengeMaterial from '../../components/ChallengeMaterial'
-import { Lesson } from '../../@types/lesson'
-import { GET_LESSONS } from '../../graphql/queries'
+
+import { GET_LESSON } from '../../graphql/queries'
 
 const Challenges: React.FC = () => {
   const router = useRouter()
-  const { loading, data } = useQuery(GET_LESSONS)
-  if (loading) {
+  const currentlessonId = router.query.lesson as string
+  const { data: sessionData, error } = useSession()
+  const userId =
+    sessionData && sessionData.userInfo
+      ? sessionData.userInfo.id.toString()
+      : ''
```

Comment:
don't fire the query (below) if there is no user available to save on bandwidth.
```
if (!userId) {
  return <h1> loading....
}
```